### PR TITLE
Improved numerical validation in Filter Events Interface 

### LINKED ISSUE #36073: Improve validation in FilterEvents dialog
A couple of issues were found during manual testing of the `Interfaces->Utility->FilterEvents` dialog for release 6.8 (#36050). It would be good to check validation generally in the dialog, but these were the specific issues found:

- [x] **Crash when trying to enter 1.25e-7 into the Filter By Log tab Minimum Value field:**

1) Open the dialog
2) In the Filter By Log tab at the bottom, attempt to enter `1.25e-7` in the `Minimum Value` field. When you type the letter `e`, Mantid should throw the following error and ask if you want to terminate the application:

```
File "C:\MantidNightlyInstall\bin\lib\site-packages\mantidqtinterfaces\FilterEvents\eventFilterGUI.py", line 445, in set_minLogValue
newminY = float(self.ui.lineEdit_5.text())
ValueError: could not convert string to float: '1.25e'
```

- [x] **Crash if using a TOF correction to sample (in the Advanced Setup tab) that is not compatible with the workspace:**

1) Open the dialog
2) Browse to the `CNCS_7860_event.nxs` file from the training data. Click the `Load` button to load the data.
3) In the Advanced Setup tab at the bottom, select `Direct` from the `TOF Correction To Sample` field (for this dataset the `Elastic` correction would have been the correct choice).
4) On the Filter By Time tab, click the `Filter` button. The following error should be thrown and you will be asked if you want to terminate the application:

```
Error in execution of algorithm FilterEvents:
Inelastic instrument detector 0 of spectrum 0 does not have EFixed
Traceback (most recent call last):
File "C:\MantidNightlyInstall\bin\lib\site-packages\mantid\simpleapi.py", line 1069, in call
algm.execute()
RuntimeError: Inelastic instrument detector 0 of spectrum 0 does not have EFixed

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "C:\MantidNightlyInstall\bin\lib\site-packages\mantidqtinterfaces\FilterEvents\eventFilterGUI.py", line 939, in filterByTime
self.splitWksp(splitws, infows)
File "C:\MantidNightlyInstall\bin\lib\site-packages\mantidqtinterfaces\FilterEvents\eventFilterGUI.py", line 1027, in splitWksp
api.FilterEvents(
File "C:\MantidNightlyInstall\bin\lib\site-packages\mantid\simpleapi.py", line 1076, in call
raise RuntimeError(msg) from e
RuntimeError: FilterEvents-v1: Inelastic instrument detector 0 of spectrum 0 does not have EFixed
```

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
Some issues were found during manual testing for the Filter Events interface in the last two release periods (#36050, #36697). A few of them were summarized in #36073. This PR fixes the issues and adds small cosmetic changes to the interface.
#### Summary of work

The numerical inputs of the Filter Events interface were `QLineEdits` validated through regular expressions. Some incorrect values (like `-` sign) were causing the interface to crash, as the signals were called with incorrect values, and the conversion to float on string `-` failed. In this PR, the numerical line edits of the interface are reimplemented by using `QDoubleValidators` classes. Additionally, slots reacting to changes in line edits are call back using `editingFinished` signals, which is only called whenever the validators return an acceptable value, an therefore prevents converting with incorrect values without adding  intermediate validation functions. Additionally, a feedback label that indicates whenever a particular input value is wrong has been added. 
Other small fixes are: 
- A rounding error in which you could set for some numbers `StopTime` larger than `StartTime` and viceversa has been fixed. 
- Some padding has been added to the borders of the interface for better visualization, also the canvas frame has been slightly reshaped. `Set` buttons are no longer necessary.
- A bug where setting `TOF Correction to Sample` would crash if setting incorrect workspace has been fixed.
- Updated manual testing instructions. 

Fixes #36073. 

### To test:
- Follow manual testing instructions: https://developer.mantidproject.org/Testing/Utility/FilterEventsInterfaceTest.html
- Filter events don't crash when following #36073 crash instructions 

_Added release notes_

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/FilterEvents/eventFilterGUI.py
Comment (GuiMacielPereira) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/FilterEvents/eventFilterGUI.py:911]: Thanks for renaming these to sensible names. It was so confusing to understand what was happening before.

**Review by GuiMacielPereira — CHANGES_REQUESTED** [review]
Overall I the changes are good and they fix the issues that were raised.

There is a big improvement in the looks of the interface and I like that you removed the 'set' button.

I left some comments that are personal taste but the main change I think should happen is to remove the validator label that shows up when I am writing inputs. I say this because:
- The label is far away from the inputs 
- The label doesn't add any new information that is not already covered by the QValidator. 
- It can be confusing when the input of minimum value is bigger than maximum value allowed and it still shows up as valid in the label (because it is a double)

I asked people in the office for alternatives and they said that the box for the input could have a red or green border, or even better to grey out the `Filter` button so the user can't start a bad Filter. I think there is no need to go that far, so I am happy if you just leave things as it is and remove the feedback label.

The only other thing I picked up is that now the minimum and maximum value need to be set in order. If for e.g. I  want to set the range `(2, 4)` to `(6, 8)` I have to change the maximum first and only then the miminum, otherwise the minimum value get overwritten. But I think the users will not encounter this very frequently so I am happy to leave as it is.

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/FilterEvents/eventFilterGUI.py
Comment (adriazalvarez) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/FilterEvents/eventFilterGUI.py]: Good idea.

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/FilterEvents/eventFilterGUI.py
Comment (adriazalvarez) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/FilterEvents/eventFilterGUI.py]: No problem.

**Comment by adriazalvarez** [comment]
Thanks for reviewing the PR. 
Yeah, I was not so sure about the label anyway. It's good to have a second opinion. I have removed it now.

**Review by GuiMacielPereira — COMMENTED** [review]
Followed through the manual testing instructions for the interface + the instructions in the issue referenced. 

- The input validation is working very well in the interface, the use of `np.clip()` is the right choice for this case. 
- The code is nice and clean, the use of `callback()` is quite smart.

The only other thing I noticed is that by following the instructions on the referenced issue, after loading the `CNCS_7860_event` workspace and setting `Advanced Setup > TOF Correction Sample` to `Direct`, `Incident Energy = 10` and `Filter by time > Time interval = 10` and clicking `Filter` does not give an error but shows the filter as successful. My understanding is that this workspace should be filtered with `Elastic` and not `Direct` so is this the expected behaviour?

**Comment by adriazalvarez** [comment]
Thanks for looking at this. 
Regarding your question: I think the behaviour you describe is validated by the algorithm, so the interface is working as intended. CNCS is a direct instrument, but you can still make elastic (I suppose for diffraction) measurements. When you try to input indirect, it doesn't work because in that case looks for the `Efixed` of the detector in the workspace metadata, which only indirect instruments have. 
You can try this script, changing from `Elastic` to `Direct` gives no problem, but `Indirect` raises error: 
```
ws = Load(Filename='CNCS_7860_event.nxs')
splitws, infows = GenerateEventsFilter(InputWorkspace=ws, UnitOfTime='Nanoseconds', LogName='SampleTemp',
        MinimumLogValue=279.9,  MaximumLogValue=279.98, LogValueInterval=0.01)

FilterEvents(InputWorkspace=ws, SplitterWorkspace=splitws, InformationWorkspace=infows,
    OutputWorkspaceBaseName='tempsplitws',
    GroupWorkspaces=True,
    FilterByPulseTime = False,
    OutputWorkspaceIndexedFrom1 = False,
    CorrectionToSample = "Direct",
    IncidentEnergy = 3, 
    SpectrumWithoutDetector = "Skip",
    SplitSampleLogs = False,
    OutputTOFCorrectionWorkspace='mock')

# Print result
wsgroup = mtd["tempsplitws"]
wsnames = wsgroup.getNames()
for name in sorted(wsnames):
    tmpws = mtd[name]
    print("workspace %s has %d events" % (name, tmpws.getNumberEvents()))
```

**Comment by GuiMacielPereira** [comment]
Brilliant! I'll approve it then

**Review by GuiMacielPereira — DISMISSED** [review]

File: docs/source/release/v6.10.0/Framework/Python/Bugfixes/36073.rst
Comment (SilkeSchomann) [inline docs/source/release/v6.10.0/Framework/Python/Bugfixes/36073.rst:2]:
```suggestion
- Fix crash occurring when an incorrect `TOF Correction To Sample` value is selected for the input workspace. Now error message is displayed instead of crash.
```

Comment (SilkeSchomann) [reply docs/source/release/v6.10.0/Framework/Python/Bugfixes/36073.rst:2]: @adriazalvarez I know I am nit-picking here!

Comment (adriazalvarez) [reply docs/source/release/v6.10.0/Framework/Python/Bugfixes/36073.rst]: @SilkeSchomann No problem!

**Review by SilkeSchomann — APPROVED** [review]
Approving again after requested change was added.

**Pull request merged**
